drivers/nrf_wifi/os_if/inc/osal_structs.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * @brief File containing structure declarations for the
+ * OSAL Layer of the Wi-Fi driver.
+ */
+
+#ifndef __OSAL_STRUCTS_H__
+#define __OSAL_STRUCTS_H__
+
+#ifdef __ZEPHYR__
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdarg.h>
+#elif __KERNEL__
+/* For Linux, use kernel internal headers instead of C headers*/
+#include <linux/stddef.h>
+#include <linux/string.h>
+#include <linux/stdarg.h>
+#endif
```

Comment:
isn't code supposed to be OS agnostic?

Reply:
Yes, this was a workaround that we did for Linux driver build as C library isn't available and we should use Linux kernel's variants. Still OS agnostic, but few places we had per-OS workarounds.